### PR TITLE
fix(backup): appropriate warning messages for credential data

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#7159

### CODE REVIEW

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@c3y1huang Can you help review? Thank you.

File: datastore/longhorn.go [hunk 1]
```diff
@@ -313,6 +313,16 @@ func (s *DataStore) ValidateSetting(name, value string) (err error) {
 		for _, checkKey := range checkKeyList {
 			if value, ok := secret.Data[checkKey]; ok {
 				if strings.TrimSpace(string(value)) != string(value) {
+					switch {
+					case strings.TrimLeft(string(value), " ") != string(value):
+						return fmt.Errorf("there is leading white space in %s", checkKey)
```
Comment (c3y1huang) [inline datastore/longhorn.go:318]:
```suggestion
						return fmt.Errorf("invalid leading white space in %s", checkKey)
```
Could you also fix the others? Thank you.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

File: datastore/longhorn.go (hunk 1 above)
Comment (mantissahz) [reply datastore/longhorn.go]: Sure. Modified. Thanks

**Review by c3y1huang — APPROVED** [review]
The value could contain mixed invalid conditions, so it's better to check all conditions and aggregate the errors with [MultiError](https://github.com/longhorn/longhorn-manager/blob/master/util/multierror.go). However, I am also fine with the current implementation, we can improve this later.

**Comment by mantissahz** [comment]
@Mergifyio backport v1.6.x

**Comment by mantissahz** [comment]
@Mergifyio backport v1.5.x

**Pull request merged**